### DISCUSSION TITLE
Incorrect routing advice of "turn right"when "straight on"was correct

### DISCUSSION THREAD

**Discussion by nm7s9** [osmandapp/OsmAnd#19659] [Maps Related]
When navigating north east away from Freemantle , Western Australia, along the Leach Highway (7) the Osmand offline routing,  advised to "turn right on 7 Leach Highway then keep left".   In fact turning right as advised resulted in turning right onto Stock road rather than the correct stay straight on onto 7 Leach highway.

This incorrect advise seemed to have been at the exact point where the trunk_link slip road deviated left off Leach Highway onto Stock road.

After turning right at the intersection, onto stock road the routing then , correctly got me to do a u_turn and then turn right back onto Leach highway.

This same error occurred again further along Leach highway at the intersection of this highway with Barbican Street, on the left, and Vahland Avenue on the right. Once again it directed me, at the point where the slip road to Barbicam Street on the left, starts,  to "turn right on 7 Leach highway" whereas it should have said (if anything), continue straight on 7 Leach highway.

Leach Highway , before Stock Road  32 degree 02' 58.3" S   115 degree 47' 30.2" E
https://www.openstreetmap.org/#map=19/-32.04991/115.79043

Leach Highway, Before  Barbican Street/Vahland Avenue  32 degree 02'04.2" S    115 degree 53'01.3 E
https://www.openstreetmap.org/#map=19/-32.03373/115.88470

Osmand+ 4.7.10  on Galaxy Z-Fold 4  offline maps and routing.
Map  Australia - Western Australia created 01.04.2024 11.00 size 185.7 MB
Routing profile was for a car

**Comment by yuriiurshuliak** [comment]
Please provide the precise coordinates of the starting point and the endpoint of your route. This will enable us to effectively reproduce and investigate the reported routing issue with incorrect instructions turns.

**Reply by nm7s9** [reply]
I have noticed this  effect at 4 intersections.

Two from Western Australia on a holiday there last week and two in Canberra ACT , yesterday.
Three of these I have been able to prevent by more precise mapping of the intersection.

I believe that the "incorrect" voice prompt (and visual turn prompt) may be caused by one , or more, of the following things.

1) The angle that the slip road comes off at from the source road
2) Whether the slip and source roads are the same priority or not
3) Whether there are lane turn taggings at the intersection or not.
4)  Whether the source road is split at the exact spot that the slip road diverges.

One of the ones in Canberra was on a major route often used by tourists, so I am happy to have fixed that one.

The one that I still can not fix is one of the ones in Western Australia  ( Leach Highway, Vahland Avenue intersection)

Start point  32 degree 1'47.3"S    115 degree 53'25.5 E
End point    32 degree 1'42.5"S    115 degree 53'32.4 E

I expect there will be a multitude of these situations in the Australian mapping so I will start searching for them using simulated routing.

**Reply by vshcherb** [reply]
Could you please create a  route here https://osmand.net/map/ to indicate which turns are not correct ? It should be same routing

**Reply by nm7s9** [reply]
Yep - have just done that....

Using the Osmand web portal and viewing the navigstion nodes for this route shows

""Turn right (C|C|C|+TR) onto Leach Highway 7 and go 0.5 km.     (AVOID) Way 295364256"

This way is the bit of the route where the slip road deviates, up to the traffix signals which are just before the intersection point.  The intersection point has a "no left turn"on it since you are meant to take the slip road, if turning left and maybe this is confusing the router.


There is an intersection that I found today in Canberra where routing straight through it in 3 of the 4 directions show this behaviour but the 4th direction works perfectly.

I'm hoping that inspection of the OSM data and what "avoids" the routing is doing may show what is causing the confusion.


PS, I absolutely love the Osmand Web portal!!!

**Reply by vshcherb** [reply]
Please share a link to save some time for others ;-)

**Reply by nm7s9** [reply]
A link to What?     A permalink to the Leach Highway, Vahland Intersection or to the one in Canberra that I will be investigating?

**Reply by nm7s9** [reply]
https://www.openstreetmap.org/way/667100458.359375#map=19/-35.41552/149.07456

Is the one in Canberra that I think will provide the answers and

https://www.openstreetmap.org/#map=19/-32.03373/115.88470

Is the Leach Highway , Vahland Intersection in Western Australia

**Reply by yuriiurshuliak** [reply]
Are you referring to this specific route? Here is the link to view it on the web version: [Route Web Version](https://test.osmand.net/map/?start=-32.029861,115.890417&end=-32.028472,115.892333&profile=car#18/-32.02912/115.89094).

**Reply by nm7s9** [reply]
Yes

**Reply by nm7s9** [reply]
On the intersection in Canberra (given above)   which is the intersection of Drakford Drive Soward Way, and Erindale drive, I noticed that out of the 4 straight through routes  (north, south east and west), only the East one didn't show this odd behaviour and that this one was the only route where the slip road (off to the left) had the tagging  (lanes:1  turn:lanes:left)

However adding this tagging (to a completely different intersection in Canberra that also shows this same behaviour, did not fix the problem, so the lack of this tagging is also not the complete answer but may be part of a complicated routing decision.

It does seem that the behaviour is associated with turn lanes at intersections that have those short "slip road"shortcut connectors.

**Reply by vshcherb** [reply]
Yes looks like a bug which we need to create & fix

**Reply by yuriiurshuliak** [reply]
By following this link, a corresponding task is created: https://github.com/osmandapp/OsmAnd/issues/19726

**Reply by nm7s9** [reply]
Is this the same issue as https://github.com/osmandapp/OsmAnd/issues/15559
